core/src/main/java/com/microsoft/applicationinsights/internal/channel/TransmissionHandlerArgs.java

```diff
@@ -0,0 +1,32 @@
+package com.microsoft.applicationinsights.internal.channel;
+
+import org.apache.http.Header;
+
+import com.microsoft.applicationinsights.internal.channel.common.Transmission;
+
+public class TransmissionHandlerArgs {
+	private String responseBody;
+	public void setResponseBody(String body) { this.responseBody = body;}
+	public String getResponseBody() { return this.responseBody;}
+	
+	
+	private TransmissionDispatcher transmissionDispatcher;
+	public void setTransmissionDispatcher(TransmissionDispatcher dispatcher) { this.transmissionDispatcher = dispatcher;}
+	public TransmissionDispatcher getTransmissionDispatcher() { return this.transmissionDispatcher;}
+	
+	private Transmission transmission;
+	public void setTransmission(Transmission transmission) { this.transmission = transmission;}
+	public Transmission getTransmission() { return this.transmission;}
+	
+	private int responseCode;
+	public void setResponseCode(int code) { this.responseCode = code;}
+	public int getResponseCode() { return this.responseCode;}
+	
+	private Throwable exception;
```

Comment:
@debugthings do we ever get a Throwable here? I think most of the time it would get an instance of Exception class. If you know any place where this could be Throwable please let me know

Reply:
From the original implementation they catch Throwable for logging. I also consume it here incase there is an Error and want to retry the transmission.

https://github.com/Microsoft/ApplicationInsights-Java/blob/1e5d72cadd24cb75b46d577ef0f8c3828fc87e19/core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java#L190-L193

Which is then added here
https://github.com/Microsoft/ApplicationInsights-Java/blob/1e5d72cadd24cb75b46d577ef0f8c3828fc87e19/core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java#L209

And then null checked here
https://github.com/Microsoft/ApplicationInsights-Java/blob/1e5d72cadd24cb75b46d577ef0f8c3828fc87e19/core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/ErrorHandler.java#L56